config/config.go

```diff
@@ -304,138 +331,52 @@ func NewNetworkAgentConfig(yamlPath string) (*AgentConfig, error) {
 	return cfg, nil
 }
 
-// mergeEnvironmentVariables applies overrides from environment variables to the process agent configuration
-func mergeEnvironmentVariables(c *AgentConfig) *AgentConfig {
-	var err error
-	if enabled, err := isAffirmative(os.Getenv("DD_PROCESS_AGENT_ENABLED")); enabled {
-		c.Enabled = true
-		c.EnabledChecks = processChecks
-	} else if !enabled && err == nil {
-		c.Enabled = false
-	}
-
-	if v := os.Getenv("DD_HOSTNAME"); v != "" {
-		log.Info("overriding hostname from env DD_HOSTNAME value")
-		c.HostName = v
+func loadEnvVariables() {
+	for envKey, cfgKey := range map[string]string{
+		"DD_PROCESS_AGENT_ENABLED":          "process_config.enabled",
+		"DD_PROCESS_AGENT_CONTAINER_SOURCE": "process_config.container_source",
+		"DD_SCRUB_ARGS":                     "process_config.scrub_args",
+		"DD_STRIP_PROCESS_ARGS":             "process_config.strip_proc_arguments",
+		"DD_PROCESS_AGENT_URL":              "process_config.process_dd_url",
+
+		// Note: this feature is in development and should not be used in production environments
+		"DD_NETWORK_TRACING_ENABLED":  "network_tracer_config.enabled",
+		"DD_NETTRACER_SOCKET":         "network_tracer_config.nettracer_socket",
+		"DD_DISABLE_TCP_TRACING":      "network_tracer_config.disable_tcp",
+		"DD_DISABLE_UDP_TRACING":      "network_tracer_config.disable_udp",
+		"DD_DISABLE_IPV6_TRACING":     "network_tracer_config.disable_ipv6",
+		"DD_COLLECT_LOCAL_DNS":        "network_tracer_config.collect_local_dns",
+		"DD_USE_LOCAL_NETWORK_TRACER": "network_tracer_config.use_local_network_tracer",
+		"DD_ENABLE_PROFILING":         "network_tracer_config.debug_profiling_enabled",
+
+		"DD_DOGSTATSD_PORT": "dogstatsd_port",
+		"DD_BIND_HOST":      "bind_host",
+		"HTTPS_PROXY":       "proxy.https",
+		"DD_PROXY_HTTPS":    "proxy.https",
+
+		"DD_LOGS_STDOUT":    "log_to_console",
+		"LOG_TO_CONSOLE":    "log_to_console",
+		"DD_LOG_TO_CONSOLE": "log_to_console",
+		"LOG_LEVEL":         "log_level", // Support LOG_LEVEL and DD_LOG_LEVEL but prefer DD_LOG_LEVEL
+		"DD_LOG_LEVEL":      "log_level",
+	} {
+		if v, ok := os.LookupEnv(envKey); ok {
+			config.Datadog.Set(cfgKey, v)
```

Comment:
I thought the goal was to make `Set` private did it changed ?

Reply:
Yeah, I thought so too. I'm not sure how we'd properly handle environment variables then.

I'm just going to follow the trace-agent's example for now and we can change in the future depending on what happens downstream.